### PR TITLE
fix: fix dolar value calculation

### PR DESCRIPTION
**Pull request by tcar121293**
fix dolar value calculation
## Related Issue Or Context

Closes: #<issue>

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation

## Checklist:

- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have ensured that all acceptance criteria (or expected behavior) from issue are met
- [ ] I have updated the documentation locally and in chainbridge-docs.
- [ ] I have added tests to cover my changes.
- [ ] I have ensured that all the checks are passing and green, I've signed the CLA bot

### CODE REVIEW

File: src/indexer/services/coinmarketcap/coinmarketcap.service.ts [hunk 1]
```diff
@@ -35,12 +35,12 @@ class CoinMarketCapService {
...
   private async getValueConvertion(amount: string, tokenSymbol: string): Promise<CoinMaketCapResponse["quote"]["USD"]["price"]> {
-    const tokenValue: string | undefined = await this.memoryCache.get(tokenSymbol)
+    const tokenValue: number | undefined = await this.memoryCache.get(tokenSymbol)
     if (tokenValue) {
-      return BigNumber(amount).times(BigNumber(tokenValue))
+      return Number(amount) * tokenValue
     }
 
-    const url = `${this.coinMarketCapUrl}/v2/tools/price-conversion?amount=${amount}&symbol=${tokenSymbol}&convert=USD`
+    const url = path.join(this.coinMarketCapUrl, `/v2/tools/price-conversion?amount=1&symbol=${tokenSymbol}&convert=USD`)
```
Comment (mpetrun5) [inline src/indexer/services/coinmarketcap/coinmarketcap.service.ts:43]: What's the difference here :sweat_smile: ?

**Review by mj52951 — APPROVED** [review]

**Review by mpetrun5 — CHANGES_REQUESTED** [review]
I don't really get what changed except typings

File: src/indexer/services/coinmarketcap/coinmarketcap.service.ts [hunk 2]
```diff
@@ -35,12 +35,12 @@ class CoinMarketCapService {
   }
 
   private async getValueConvertion(amount: string, tokenSymbol: string): Promise<CoinMaketCapResponse["quote"]["USD"]["price"]> {
-    const tokenValue: string | undefined = await this.memoryCache.get(tokenSymbol)
+    const tokenValue: number | undefined = await this.memoryCache.get(tokenSymbol)
     if (tokenValue) {
-      return BigNumber(amount).times(BigNumber(tokenValue))
+      return Number(amount) * tokenValue
```
Comment (mpetrun5) [inline src/indexer/services/coinmarketcap/coinmarketcap.service.ts:40]: Why did we change this from BigNumber

Comment (tcar121293) [reply src/indexer/services/coinmarketcap/coinmarketcap.service.ts:40]: I will revert the change. The coinmarketcap returns the value of type number, that's why I changed everything into number. But then again, the amount can be BigNumber...

File: src/indexer/services/coinmarketcap/coinmarketcap.service.ts (hunk 1 above)
Comment (tcar121293) [reply src/indexer/services/coinmarketcap/coinmarketcap.service.ts:43]: amount=1 in the url. If we pass the actual value of the amount, the coinmarketcap automatically calculates the usdValue. But then we can't store it in memory to use it later to reduce the number of coinmarketcap API calls

**Review by MakMuftic — APPROVED** [review]

**Review by mpetrun5 — APPROVED** [review]

**Pull request merged**